src/surface.rs

```diff
@@ -47,6 +47,61 @@ impl Surface {
                                                                   height))
         }
     }
+
+    pub fn get_mime_data(&self, mime_type: &str) -> Option<&[u8]> {
+        let mut data_ptr: *mut u8 = ptr::null_mut();
+        let mut length = 0u64;
+        unsafe {
+            ffi::cairo_surface_get_mime_data(
+                self.to_raw_none(),
+                mime_type.to_glib_none().0,
+                &mut data_ptr,
+                &mut length,
+            );
+            if !data_ptr.is_null() && length != 0 {
+                Some(slice::from_raw_parts(
+                    data_ptr as *const u8,
+                    length as usize,
+                ))
+            } else {
+                None
+            }
+        }
+    }
+
+    pub fn set_mime_data<T: AsRef<[u8]> + Send + 'static>(&self,
+                                                          mime_type: &str,
+                                                          slice: T) -> Result<(), Status> {
+        let b = Box::new(slice);
+        let (size, data) = {
+            let slice = (*b).as_ref();
+            (slice.len(), slice.as_ptr())
+        };
+
+        let user_data = Box::into_raw(b);
+
+        let status = unsafe {
+            ffi::cairo_surface_set_mime_data(
+                self.to_raw_none(),
+                mime_type.to_glib_none().0,
+                data,
+                size as u64,
+                Some(unbox::<T>),
+                user_data as *mut _,
+                )
```

Comment:
Please make indent smaller

Reply:
How wide should the indent be? Isn't it 4 spaces?

Reply:
4 spaces, yes

Reply:
That's why we really need rustfmt ;)

Reply:
i really want rustfmt.

Reply:
First comment was about closing bracket has same indent as parameters.

Reply:
I hope you are happy with the last change @EPashkin ;)

Reply:
Yes, thanks 😉